### PR TITLE
docs/user/versioning: Document APIs covered by our Semantic Versioning

### PR DESCRIPTION
**Pull request by wking**
Also roll the tips-and-tricks docs into the overview's multi-invocation docs, since they'd been covering the same ground with slightly different wording before.  I've expanded the unified description to go into a bit more detail and tie in the new versioning docs.

I've also documented the `manifest-templates` target from 166a9f1e (#592).

CC @smarterclayton, @derekwaynecarr, @eparis.

### CODE REVIEW

File: docs/user/overview.md [hunk 1]
```diff
@@ -47,6 +50,38 @@ The following targets can be destroyed by the installer:
 
 ### Multiple Invocations
 
-In order to allow users to customize their installation, the installer can be invoked multiple times. The state is stored in a hidden file in the target directory and contains all of the intermediate artifacts. This allows the installer to pause during the installation and wait for the user to modify intermediate artifacts.
+In order to allow users to customize their installation, the installer can be invoked multiple times. The state is stored in a hidden file in the asset directory and contains all of the intermediate artifacts. This allows the installer to pause during the installation and wait for the user to modify intermediate artifacts.
```
Comment (wking) [inline docs/user/overview.md]: Changing "target" -> "asset" here gets us closer to [the wording we use for `--dir`][1].

[1]: https://github.com/openshift/installer/blob/v0.7.0/cmd/openshift-install/main.go#L46

File: docs/user/overview.md [hunk 2]
```diff
@@ -47,6 +50,38 @@ The following targets can be destroyed by the installer:
...
+cp install-config.yaml cluster-0/
+openshift-install --dir=cluster-0 create cluster
+```
+
+Supplying a previously-generated install-config like this is [explicitly part of the stable API](versioning.md).
+
+You can also edit the assets in the asset directory during a single run.
+For example, you can adjust [the cluster-version operator's configuration][cluster-version]:
+
+```sh
```
Comment (abhinavdahiya) [inline docs/user/overview.md]: So personally i thought the reasoning for manifests templates was the a advanced user knows the `variables` that are supported in the templates and maybe change to add new or remove some...

maybe include example of editing the `manifests`

**Comment by abhinavdahiya** [comment]
> Also roll the tips-and-tricks docs into the overview's multi-invocation docs, since they'd been covering the same ground with slightly different wording before.

^^ @crawford

File: docs/user/overview.md (hunk 2 above)
Comment (wking) [reply docs/user/overview.md]: > maybe include example of editing the `manifests`

Good idea.  Fixed with e2f8361 -> a017e43.

**Comment by crawford** [comment]
Can you rebase this so we can get this in?

**Comment by wking** [comment]
> Can you rebase this so we can get this in?

Rebased onto master with a017e43 -> 8df754a.

**Comment by wking** [comment]
Prow seemed hung ("shellcheck Expected — Waiting for status to be reported" and similar), so I've pushed 8df754a -> 28592d0 with a new commit date.

**Comment by wking** [comment]
Finally all green :).

**Pull request merged**